### PR TITLE
feat(action-menu): Adds setFocus method and focus on first action when open or menu button when closed.

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #1896

## Summary

feat(action-menu): Adds setFocus method and focus on first action when open or menu button when closed.

### CODE REVIEW

File: src/components/calcite-action-menu/calcite-action-menu.tsx [hunk 1]
```diff
@@ -142,6 +143,17 @@ export class CalciteActionMenu {
...
 
+  // --------------------------------------------------------------------------
+  //
+  //  Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method()
+  async setFocus(): Promise<void> {
+    focusElement(this.open ? this.actionElements[0] || this.menuEl : this.menuButtonEl);
```
Comment (jcfranco) [inline src/components/calcite-action-menu/calcite-action-menu.tsx:154]: Would there be cases where we want a `focusId` for the menu button?

File: src/components/calcite-action-menu/calcite-action-menu.tsx [hunk 2]
```diff
@@ -274,11 +285,14 @@ export class CalciteActionMenu {
...
-      (this.el
-        .querySelector("slot")
-        ?.assignedElements({ flatten: true })
-        .filter((el) => el.tagName === "CALCITE-ACTION") as HTMLCalciteActionElement[]) || [];
+    const { el } = this;
+
+    const assignedElements = el
+      .querySelector("slot")
+      ?.assignedElements({ flatten: true })
+      .filter((element) => element?.tagName === "CALCITE-ACTION") as HTMLCalciteActionElement[];
```
Comment (jcfranco) [inline src/components/calcite-action-menu/calcite-action-menu.tsx:293]: `element?.tagName` shouldn't need optional chaining.

**Review by jcfranco — APPROVED** [review]
👍🏼

File: src/components/calcite-action-menu/calcite-action-menu.tsx (hunk 1 above)
Comment (driskull) [reply src/components/calcite-action-menu/calcite-action-menu.tsx:154]: I don't think so. It depends on if it's open or not so we can go with that for now.

**Pull request merged**